### ISSUE TITLE
[Node] Reload config file every time it tries to mine a block

### ISSUE THREAD

**Issue by jcnelson** [stacks-network/stacks-core#4047]
The miner thread should load the miner's `satoshis_per_byte` field from the config file every time it builds a block.  Then, miners can adjust their fee rates on-the-fly without restarting their nodes.  They can even write a separate script that polls a 3rd party fee-rate service and writes this value to the config file.

**Comment by AcrossfireX** [comment]
In order to make best use of the PR that would mitigate this issue, we need a system of determining what the appropriate fee is and also when to make an update to the config file so this does not need to be manually done by miners each step of the way as fees change. This can be done in a few steps that can be done by a script run periodically within cron.

1. Make an API request to a destination to get the latest fee estimation, this may be an electrum server or alternative endpoint that has the appropriate data.
2. Store that result as a variable
3. Use sed in bash to update the miner config file to have the appropriate value.

Once those are done, the manual process of updating is complete and the miner should not have to do anything to keep up to date with fee market dynamics.

Here is an example script that does this (this may need to be modified for your particular scenario).
```
#!/bin/bash
#REPLACE THIS PATH WITH YOUR MINER CONFIG FILE
MINER_CONFIG=/var/tmp/config


fee_estimate=`curl https://api.blockcypher.com/v1/btc/main | grep high_fee_per_kb | awk -F ':' '{print $2}' | cut -d',' -f 1 | xargs`

sed -i "s/satoshis_per_byte =.*/satoshis_per_byte = ${fee_estimate}/g" $MINER_CONFIG
echo "fee estimate updated"
```

Users may want to modify the endpoint used for fee estimates to use their own node and also may want to add additional logic to maybe only update the fees if its a significant enough change from what they already had set, but this may be helpful nonetheless

**Comment by jcnelson** [comment]
This issue was closed by #4061.  I've moved @AcrossfireX's comment to a separate issue.
